src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -139,9 +157,18 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
                     {
                         // Don't use the individual CT here, this is a shared operation that shouldn't be affected by an individual's cancellation.
                         // The transport should have it's own timeouts, etc..
-                        _currentConfiguration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
+                        var configuration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
                         _lastRefresh = now;
```

Comment:
Does it make more sense to set the _currentConfiguration only if:
1. no validator is provided
2. validation succeeds, or validation fails but _currentConfiguration is null


                        var configuration = await _configRetriever.GetConfigurationAsync(MetadataAddress, _docRetriever, CancellationToken.None).ConfigureAwait(false);
                        _lastRefresh = now;
                        _syncAfter = DateTimeUtil.Add(now.UtcDateTime, AutomaticRefreshInterval);
                        if (_configValidator != null)
                        {
                            ConfigurationValidationResult result = _configValidator.Validate(configuration);
                            if (!result.Succeeded)
                            {
                                LogHelper.LogWarning(LogMessages.IDX20810, result.Exception);
                                if (_currentConfiguration == null)
                                    _currentConfiguration = configuration;
                            }
                            else
                            {
                                _currentConfiguration = configuration;
                            }
                        }
                        else
                        {
                            _currentConfiguration = configuration;
                        }